app/services/audiobook/tts.py

```
@@ -0,0 +1,99 @@
+import requests
+from abc import ABC, abstractmethod
+from typing import Any
+import re
+
+class TTSProvider(ABC):
+    name: str = "base"
+    supported_formats: set[str] = {"mp3", "wav"}
```

Ruff (0.15.17)

[warning] Mutable default value for class attribute

(RUF012)